src/odemis/acq/stitching/_registrar.py

```diff
@@ -320,180 +318,135 @@ def _estimateMatch(self, imageA, imageB, shift=(0, 0)):
-        pos = None
-        match = 0
+        if (xdir == LEFT_TO_RIGHT and col == 0) or (xdir == RIGHT_TO_LEFT and col == self.nx):
+            return None, 0
 
+        # Compute shift
         tile = self.tiles[row][col]
-        # the expected shift vector between the two tiles if the scanning would
-        # be ideal
-        exp_shift = (self.tsize, 0)
-
-        if (xdir == LEFT and col > 0) or (xdir == RIGHT and col < self.nx):
-            if xdir == LEFT:
-                prev_pos = self.shifts[row][col - 1]
-                prev_tile = self.tiles[row][col - 1]
-                x, y = self._get_shift(prev_tile, tile, exp_shift)
-            elif xdir == RIGHT:
-                prev_pos = self.shifts[row][col + 1]
-                prev_tile = self.tiles[row][col + 1]
-                x, y = self._get_shift(tile, prev_tile, exp_shift)
-                y = -y
-                x = -x
-            match = self._estimateMatch(
-                prev_tile, tile, (exp_shift[0] - x, exp_shift[1] - y))
-
-            if match < BEST_MATCH and (row > 1 or col > 1):
-                # if the shift calculated does not give a sufficient match,
-                # try the horizontal mean shift along this row and column if
-                # possible
-                try:
-                    mean_x, mean_y = self._mean_shift(row, col, self.coords_hor)
-                except ValueError:
-                    # if a mean value is not available try the last horizontal
-                    # shift measured
-                    mean_x, mean_y = self.last_hor
-                match_c = self._estimateMatch(prev_tile, tile,
-                                              (exp_shift[0] - mean_x, exp_shift[1] - mean_y))
-                if match_c > match:
+        if xdir == LEFT_TO_RIGHT:
+            exp_shift = (int(self.size[1] - self.osize), 0)
+            prev_pos = self.registered_positions[row][col - 1]
+            prev_tile = self.tiles[row][col - 1]
+        elif xdir == RIGHT_TO_LEFT:
+            exp_shift = (-int(self.size[1] - self.osize), 0)
+            prev_pos = self.registered_positions[row][col + 1]
+            prev_tile = self.tiles[row][col + 1]
+        else:
+            raise ValueError("xdir argument is %s, must be either LEFT_TO_RIGHT or RIGHT_TO_LEFT." % xdir)
+        x, y = self._get_shift(prev_tile, tile, exp_shift)
+
+        # If the quality of the cross-correlation is low, use fallback shift
+        match = self._estimateMatch(prev_tile, tile, (exp_shift[0] - x, exp_shift[1] - y))
+        if match < MIN_MATCH and (row > 1 or col > 1):
+            logging.debug('Horizontal registration results in bad match. Using fallback position.')
+            # Use mean shift of all previously registered tiles in horizontal direction. If none are
+            # available, fall back to expected position (i.e. zero shift)
+            if filter(None, self.shifts_hor[row]) != []:
```

Comment:
Avoid using `l != []`, as you can just write `l`.
Here you could _maybe_ write more clearly `if any(s is not None for s in  self.shifts_hor[row]):`

Alternatively, you could also just pre-compute it, and by giving it a variable name it'd also clarify the code. Something like:
```
known_shifts_x = filter(None, self.shifts_hor[row])
if known_shifts_x:
   mean_x = numpy.mean(known_shifts_x)
:
```